evap/staff/views.py

```diff
@@ -1207,6 +1207,15 @@ def notify_reward_points(grantings, **_kwargs):
 
         return redirect("staff:semester_view", semester.id)
 
+    questionnaires_with_answers_per_contributor = {}
+    for contribution in evaluation.contributions.all():
+        questionnaires = Questionnaire.objects.filter(
+            Q(questions__textanswer__contribution=contribution)
+            | Q(questions__ratinganswercounter__contribution=contribution)
+        ).distinct()
+        if questionnaires.count() > 0:
+            questionnaires_with_answers_per_contributor[contribution.contributor] = questionnaires
```

Comment:
Yes. In general, neither `qs.count()` nor `qs.exists()` fill the query set cache because they are implemented at the database level without actually fetching records. `len(qs)` and `bool(qs)` do fetch the records and fill the cache.

Here, as a micro-optimization, we could do
```suggestion
        if len(questionnaires) > 0:
            questionnaires_with_answers_per_contributor[contribution.contributor] = questionnaires
```